services/src/main/scala/cromwell/pipeline/service/ProjectService.scala

```diff
@@ -14,13 +14,13 @@ class ProjectService(projectRepository: ProjectRepository, projectVersioning: Pr
 ) {
 
   private[service] def getUserProjectById(projectId: ProjectId, userId: UserId): Future[Project] =
-    projectRepository.getProjectById(projectId).flatMap(getForUserOrFail(_, userId))
+    projectRepository.getProjectById(projectId).flatMap(project => getForUserOrFail(project.toSeq, userId))
 
   def getUserProjectByName(namePattern: String, userId: UserId): Future[Project] =
-    projectRepository.getProjectByName(namePattern).flatMap(getForUserOrFail(_, userId))
+    projectRepository.getProjectsByName(namePattern).flatMap(getForUserOrFail(_, userId))
 
-  private def getForUserOrFail(project: Option[Project], userId: UserId): Future[Project] =
-    project match {
+  private def getForUserOrFail(projects: Seq[Project], userId: UserId): Future[Project] =
+    projects.find(_.ownerId == userId).orElse(projects.headOption) match {
       case Some(project) if project.ownerId == userId => Future.successful(project)
```

Comment:
Much better than before, but I think we can do even better. We should be able to get rid of double check `ownerId == userId` and weird `projects.find().orElse(projects.headOption)`

Reply:
@myazinn done